src/main/scala/ztis/ProcessTweetTask.scala

```diff
@@ -0,0 +1,36 @@
+package ztis
+
+import com.typesafe.scalalogging.slf4j.StrictLogging
+
+import scalaj.http.HttpOptions
+
+class ProcessTweetTask(cassandraClient: CassandraClient, tweet: Tweet) extends Runnable with StrictLogging {
+  
+  override def run(): Unit = {
+    val links = TweetURLExtractor.extractLinks(tweet)
+
+    val resolvedLinks =   links.flatMap(followLink).map(java.net.URI.create(_))
```

Comment:
Maybe - but during tweet fetching from Kafka we are already scheduling async task (ProcessTweetTask to be exact) so I don't know if using future here will have many benefits.